students/ChelseaSmith/lesson9/mailroom.py

```diff
@@ -0,0 +1,113 @@
+# object oriented mailroom program
+# Donor class holds individual donor information
+# Manager class accesses a text file storing all donors and manages that info
+
+class Donor:
+
+    def __init__(self, name, *args):
+        self._name = name
+        self._donations = list(args)
+
+    def append(self, addon):
+        self._donations.append(addon)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def donations(self):
+        return self._donations
+
+    def total_given(self):
+        total = sum(self._donations)
+        return total
+
+    def number_of_gifts(self):
+        number = len(self._donations)
+        return number
+
+    def average_gift(self):
+        ave = sum(self._donations)/len(self._donations)
+        return ave
+
+    def write_letter(self):
+        letter = "Dear {},\nThank you for your generous donation of ${:.2f}.\nSincerely,\n Generic Charities".format(self._name, self._donations[-1])
+        return letter
+
+
+class Manager:
+
+    def __init__(self, database):
+        self.database = database
+        self.record = list()
+        for line in open(database):
+            self.record.append(Donor(line))
```

Comment:
Couldn't get the code to run but curious how this works? The code appears to save a line with name and donations, but this appears to only pass in one arg which would map to the name param?